internal/user/user_manager_test.go

```diff
@@ -51,16 +51,66 @@ func TestUserManager(t *testing.T) {
 		t.Errorf("unexpected password hash: %v, want %v", got, want)
 	}
 
-	err = user.DeleteUserProfile(ctx, env.RepositoryWriter, "alice")
+	err = user.DeleteUserProfile(ctx, env.RepositoryWriter, "alice@somehost")
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
 
-	if _, err = user.GetUserProfile(ctx, env.RepositoryWriter, "alice"); !errors.Is(err, user.ErrUserNotFound) {
+	if _, err = user.GetUserProfile(ctx, env.RepositoryWriter, "alice@somehost"); !errors.Is(err, user.ErrUserNotFound) {
 		t.Fatalf("unexpected error: %v", err)
 	}
 }
 
+func TestValidateUsername_Valid(t *testing.T) {
+	cases := []string{
+		"foo@bar",
+		"foo.foo@bar",
+		"foo.foo@bar.bar",
+		"foo_foo@bar",
+		"foo_foo@bar_bar",
```

Comment:
Maybe add a case with a combination of `.` and `_` in the hostname. By inspection the RE allows it.

```suggestion
		"foo_foo@bar_bar",
		"foo_foo@bar_bar.baz",
		"foo_foo@bar.bar_baz",
		"foo_foo@bar.bar.baz",
```